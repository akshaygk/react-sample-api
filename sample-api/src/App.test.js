import React from 'react'
import { shallow } from 'enzyme';
import App from './App';

let wrapped = shallow(<App/>);
describe('App component home page', () => {

  it('should render the App Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('Navbar to contain title', () => { 
    expect(wrapped.find('Link').at(0).text()).toEqual("Home");
  });
});