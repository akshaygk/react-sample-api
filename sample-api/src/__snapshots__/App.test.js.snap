// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component home page should render the App Component correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <BrowserRouter>
    <WithStyles(ForwardRef(AppBar))
      position="static"
    >
      <WithStyles(ForwardRef(Toolbar))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="open drawer"
          className="makeStyles-menuButton-2"
          color="inherit"
          edge="start"
        >
          <Memo(ForwardRef(MenuIcon)) />
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-3"
          noWrap={true}
          variant="h6"
        >
          Akshay's Shop
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-3"
          noWrap={true}
          variant="h6"
        >
          <Link
            className="makeStyles-link-8"
            to="/"
          >
            Home
          </Link>
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-3"
          noWrap={true}
          variant="h6"
        >
          <Link
            className="makeStyles-link-8"
            to="/bookmarked"
          >
            Bookmarked
          </Link>
        </WithStyles(ForwardRef(Typography))>
        <div
          className="makeStyles-search-4"
        >
          <div
            className="makeStyles-searchIcon-5"
          >
            <Memo(ForwardRef(SearchIcon)) />
          </div>
          <WithStyles(ForwardRef(InputBase))
            classes={
              Object {
                "input": "makeStyles-inputInput-7",
                "root": "makeStyles-inputRoot-6",
              }
            }
            inputProps={
              Object {
                "aria-label": "search",
              }
            }
            placeholder="Searchâ€¦"
          />
        </div>
      </WithStyles(ForwardRef(Toolbar))>
    </WithStyles(ForwardRef(AppBar))>
    <nav
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    />
    <div>
      <Switch>
        <Route
          path="/product/:id"
        >
          <Product />
        </Route>
        <Route
          path="/bookmarked/"
        >
          <Bookmarked />
        </Route>
        <Route
          path="/"
        >
          <Home />
        </Route>
      </Switch>
    </div>
  </BrowserRouter>
</div>
`;
